packages/catalog-process/src/routers/EServiceRouter.ts

```diff
@@ -132,20 +137,48 @@ const eservicesRouter = (
           limit
         );
 
+        // TODO : fix returnn types and remove this filtering
+        const validConsumers = consumers.results.reduce(
```

Comment:
@MalpenZibo ho avuto problemi sui tipi creati nella risposta di questa API, visto che ci sono state modifiche su altre PR, ho preferito aggiungere questo filtro anzichè modificare la definizione dei tipi ed avere potenziali conflitti di merge, magari può dargli un occhio. 
Questo fix è temporaneo e mirato ad avere del codice compilabile per eseguire test sulla CI

Reply:
qui potremmo creare un convertitore `consumerToApiConsumer` che piazzerei in `apiConverter`. Che problemi stavi avendo?

```
const validConsumers = consumers.results
    .filter(
        (c) =>
              c.descriptorVersion !== undefined &&
              c.consumerName !== undefined &&
              c.consumerExternalId !== undefined
        )
        .map(consumerToApiConsumer);

....

export const consumerToApiConsumer = (
  consumer: Consumer
): z.infer<typeof api.schemas.EServiceConsumer> => ({
  descriptorVersion: parseInt(consumer.descriptorVersion, 10),
  descriptorState: descriptorStateToApiEServiceDescriptorState(
    consumer.descriptorState
  ),
  agreementState: agreementStateToApiAgreementState(consumer.agreementState),
  consumerName: consumer.consumerName,
  consumerExternalId: consumer.consumerExternalId,
});
```

Reply:
i problemi principali erano dovuti alla al fatto che  descriptorVersion, consumerName e consumerExternalId risultavano poter essere valorizzati come `undefined` nell'oggetto di dominio Consumer mentre nella response delle API `EServiceConsumer`  erano esclusivamente `string`.

Reply:
la fix era nata per gestire un errore provocato da un'incongruenza tra i tipi autogenerati mediante spec OpenAPI e quelli definiti nel package comune `models` che facevano riferimento allo stesso oggetto `Consumer`; tuttavia il  dockerfile nel momento della fix differiva da quello definitivo, in dettaglio sono cambiati i comandi di copia delle folder, build ed esecuzione, ragion per cui i tipi delle rispose API non erano  autogenerati ogni volta. 
effettuando diverse prove in locole il problema non si è ripresentato.

Reply:
Fix rimossa con il commit [Remove temporary fix](https://github.com/pagopa/interop-be-monorepo/pull/37/commits/87b9a5e1b77e79ea85f10489a08664100151b040)